internal/pkg/service/buffer/storage/file.go

```diff
@@ -0,0 +1,57 @@
+package storage
+
+import (
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/local"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/staging"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/storage/target"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/key"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/model/column"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/utctime"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+const (
+	FileTypeCSV = "csv"
+)
+
+// File represents a file prepared in the staging storage to be imported into the target storage.
+// File consists from zero or more Slices.
+type File struct {
+	FileKey
+	Retryable
```

Comment:
Zvysok PR je su modelove struktury a testy pre validacne pravidla, vid adresarovu strukturu v kometári PR.

Nieco z toho sme uz mali, testy pre validacne pravidla neboli vobec (vdaka podrobnejsej validacii sa jednoduchsie debuguje nejaka chyba v logike apky).

Ako sme riesili uz niekde skor, tento model pre doménu `storage` je oddeleny od modelu pre domenu `definition`.
Su tak oddelene docasne objekty (`storage` files/slices) a trvale objekty z API (`definition`, receiver/mapping/...).

Predtym to bolo spolu a robilo to problemy.